code/modules/mob/living/simple_animal/abnormality/waw/slitcurrent.dm

```diff
@@ -0,0 +1,246 @@
+//Dream-Devouring Slitcurrent would be a breach that forces you to either try to kill it from it hitting the tubes or using the tubes to make you last longer.
+//When out you're on a timer due to everyone on the floor it's on taking oxygen damage.
+
+/mob/living/simple_animal/hostile/abnormality/slitcurrent
+	name = "\proper Dream-Devouring Slitcurrent"
+	desc = "An abnormality resembling a giant black and teal shark or fish. \
+	There's teal light tubes embedded in its body,"
+	icon = 'ModularTegustation/Teguicons/64x64.dmi'
+	icon_state = "slitcurrent"
+	icon_living = "slitcurrent"
+	pixel_x = -16
+	base_pixel_x = -16
+	move_to_delay = 3
+	rapid_melee = 2
+	melee_damage_lower = 25
+	melee_damage_upper = 35
+	melee_damage_type = RED_DAMAGE
+	maxHealth = 3000
+	health = 3000
+	damage_coeff = list(RED_DAMAGE = 0.5, WHITE_DAMAGE = 0.5, BLACK_DAMAGE = 0.5, PALE_DAMAGE = 0.5)
+	stat_attack = HARD_CRIT
+	deathsound = 'sound/abnormalities/dreamingcurrent/dead.ogg'
+
+	attack_verb_continuous = "chomps"
+	attack_verb_simple = "chomps"
+
+	threat_level = WAW_LEVEL
+
+	work_chances = list(
+						ABNORMALITY_WORK_INSTINCT = list(85, 65, 45, 50, 55),
+						ABNORMALITY_WORK_INSIGHT = 45,
+						ABNORMALITY_WORK_ATTACHMENT = -100,//you thought it would work like current eh?
+						ABNORMALITY_WORK_REPRESSION = list(70, 65, 60, 55, 50)//Incase you reach waw with justice 1
+						)
+	work_damage_amount = 6//does constant oxygen damage during work
+	work_damage_type = RED_DAMAGE
+	ranged = 1
+	can_breach = TRUE
+	start_qliphoth = 2
+	can_patrol = TRUE
+
+	ego_list = list(
+		/datum/ego_datum/weapon/dream_devouring,
+		/datum/ego_datum/weapon/blind_obsession,
+		/datum/ego_datum/armor/blind_obsession
+		)
+	gift_type = /datum/ego_gifts/blind_obsession
+	abnormality_origin = ABNORMALITY_ORIGIN_LIMBUS
+
+	var/stunned = FALSE
+	//Stuff relating to the dive attack
+	var/diving = FALSE
+	var/dive_cooldown
+	var/dive_cooldown_time = 15 SECONDS
+	var/dive_damage = 100
+	var/flotsam = FALSE
+	//The amount of flotsams that should spawn in the hallways when it breaches
+	var/tube_spawn_amount = 6
+	//Stuff relating to the facility wide oxygen damage
+	var/oxy_cooldown
+	var/oxy_cooldown_time = 5 SECONDS
+
+/mob/living/simple_animal/hostile/abnormality/slitcurrent/Initialize()
+	. = ..()
+	color = COLOR_TEAL
+
+/mob/living/simple_animal/hostile/abnormality/slitcurrent/Life()
+	. = ..()
+	if(!.) // Dead
+		return FALSE
+	if(oxy_cooldown_time <= world.time)
+		OxygenLoss()
```

Comment:
If you want this every 5 seconds, then you should probably put it on a looping timer instead of life, which only ticks on 2 second intervals.